pkg/controller/kubedirectorconfigmap/kubedirectorconfigmap_controller.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2019 Hewlett Packard Enterprise Development LP
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubedirectorconfigmap
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bluek8s/kubedirector/pkg/shared"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_kubedirectorconfigmap")
+
+// Add creates a new KubeDirectorConfigMap Controller and adds it to the Manager.
+// The Manager will set fields on the Controller and Start it when the Manager
+// is Started.
+func Add(mgr manager.Manager) error {
+
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler.
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+
+	return &ReconcileKubeDirectorConfigMap{scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler.
+func add(
+	mgr manager.Manager,
+	r reconcile.Reconciler,
+) error {
+
+	// Create a new controller
+	c, err := controller.New("kubedirectorconfigmap-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource KubeDirectorConfigMap.
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileKubeDirectorConfigMap implements
+// reconcile.Reconciler.
+var _ reconcile.Reconciler = &ReconcileKubeDirectorConfigMap{}
+
+const (
+	// Period between the time when the controller requeues a request and
+	// it's scheduled again for reconciliation. Zero means don't poll.
+	// For now we don't need polling on this CR. Some anticipated features
+	// will need polling at which point we can change this.
```

Comment:
Should remove the comment about "some anticipated features" (unless it is true!).

Do we really not need polling in this case? I guess I can see that.

Reply:
Also, to be picky, this is not a CR (custom resource). Just a (standard) resource!

I have the feeling I'm also going to see "CR" in a lot of the new variable names that represent a configmap. Ideally we'd change those to just say configmap instead. Don't know how much of a hassle that will be.

Reply:
Yes, makes sense